merlion/models/utils/rolling_window_dataset.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+from typing import Optional
+from merlion.utils.time_series import TimeSeries
+
+
+class RollingWindowDataset:
+
+    def __init__(
+            self,
+            data: TimeSeries,
+            target_seq_index: int,
+            maxlags: int,
+            forecast_steps: int,
+            shuffle: bool = False,
+            ts_index: bool = False,
+            batch_size: Optional[int] = 1,
+            label_axis: int = 0,
+    ):
+        assert isinstance(data, TimeSeries), \
+            "RollingWindowDataset expects to receive TimeSeries data"
+        data = data.align()
+        self.dim = data.dim
+        self.batch_size = batch_size
+        self.ts_index = ts_index
+
+        if ts_index:
+            self.data_ts = data
+            self.target_ts = data.univariates[data.names[target_seq_index]]
+        else:
+            data_df = data.to_pd()
+            self.data = data_df.values
+            self.timestamp = data_df.index
+            self.target = data.univariates[data.names[target_seq_index]].values
+            self.target_timestamp = data.univariates[data.names[target_seq_index]].index
+
+        self.target_seq_index = target_seq_index
+        self.maxlags = maxlags
+        self.forecast_steps = forecast_steps
+
+        self.shuffle = shuffle
+        self.label_axis = label_axis
+
+        self._valid_rolling_steps = len(data) - self.maxlags - self.forecast_steps
+        self._data_len = len(data)
+
+    @property
+    def valid_rolling_steps(self):
+        return self._valid_rolling_steps
+
+    def __len__(self):
+        return self._data_len
```

Comment:
1. Why not just directly return `len(self.data) - self.maxlags - self.forecast_steps` from `valid_rolling_steps`?  It seems unnecessary to add an intermediate variable.
2. To me, `valid_rolling_steps` actually makes more sense as `len(self)`, since it's the number of items in the iterator.

Reply:
remove the intermediate variable, and depending on the iteration method, calculate the bound in the related functions